src/entities/nestedExit/encodeCalls.ts

```diff
@@ -0,0 +1,119 @@
+import { Hex } from '../../types';
+import { WeightedEncoder } from '../encoders';
+import { ComposableStableEncoder } from '../encoders/composableStable';
+import { NestedExitCallAttributes } from './types';
+import { Relayer } from '../relayer';
+import { replaceWrapped } from '../utils/replaceWrapped';
+import { bathcRelayerLibraryAbi } from '../../abi';
+import { encodeFunctionData } from 'viem';
+
+export const encodeCalls = (
+    callsAttributes: NestedExitCallAttributes[],
+    isProportional: boolean,
+) => {
+    const encodedCalls: Hex[] = [];
+    for (const callAttributes of callsAttributes) {
+        const {
+            useNativeAssetAsWrappedAmountOut,
+            chainId,
+            sortedTokens,
+            poolId,
+            poolType,
+            kind,
+            sender,
+            recipient,
+            bptAmountIn,
+            minAmountsOut,
+            toInternalBalance,
+            outputReferenceKeys,
+            tokenOutIndex,
+        } = callAttributes;
+
+        // replace wrapped token with native asset if needed
+        let tokensOut = [...sortedTokens];
+        if (useNativeAssetAsWrappedAmountOut) {
+            tokensOut = replaceWrapped([...sortedTokens], chainId);
+        }
+
+        const _bptAmountIn = bptAmountIn.isRef
+            ? Relayer.toChainedReference(bptAmountIn.amount)
+            : bptAmountIn.amount;
+
+        let userData: Hex;
+        let outputReferences: { index: bigint; key: bigint }[] = [];
+
+        if (isProportional) {
```

Comment:
I do think it could be useful to further modularise this code by having separate functions for proportional/singleToken specific operations. Would make it easier to read because we know what those functions should be doing. WDYT?

Reply:
Agree 👍